src/imitation/algorithms/preference_comparisons.py

```diff
@@ -767,17 +735,23 @@ def _loss(
         preferences_th = th.as_tensor(preferences, dtype=th.float32)
         ground_truth = (preferences_th > 0.5).float()
         accuracy = (predictions == ground_truth).float().mean()
-        self.logger.record("accuracy", accuracy.item())
-        return th.nn.functional.binary_cross_entropy(probs, preferences_th)
+        return {
+            "loss": th.nn.functional.binary_cross_entropy(probs, preferences_th),
+            "metrics": {"accuracy": accuracy.detach().cpu()},
+        }
 
-    def _rewards(self, transitions: Transitions) -> th.Tensor:
-        preprocessed = self.model.preprocess(
+    def _rewards(
```

Comment:
`predict_th` evaluates the reward network with `no_grad` set. We need gradients on the reward function since this is a loss. Thus `predict_th` cannot be used here.